src/utils/chat/responseGenerator.ts

```
@@ -0,0 +1,363 @@
+  }
+
+  private mergeToolCallDeltas(
+    toolCalls: ToolCallDelta[],
+    existingToolCalls: Record<number, ToolCallDelta>,
+  ): Record<number, ToolCallDelta> {
+    return toolCalls.reduce(
+      (merged, toolCall) => {
+        const { index } = toolCall
+
+        if (!merged[index]) {
+          return { ...merged, [index]: toolCall }
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)